src/core/core/rpc_engine.cpp

```diff
@@ -144,7 +144,12 @@ namespace dsn {
                 req->server_address.group_address()->leader_forward();
             }
 
-            call->set_delay(delay_ms);
+            // the "delay_ms" is simulator's net_delay_milliseconds,
+            // the "call->delay_milliseconds()" is user specified delay time to dispatch response task,
```

Comment:
I'm not sure about your "sum" case. For RPC response tasks, upper apps do not specify the delay for them (they should be executed immediately when the reply is received or timeout happened). The delay_ms is for tooling purpose in this case such as network delay in simulator or by fault injection.